test/metric/metric_value_query.go

```diff
@@ -53,17 +59,34 @@ type MetricValueFetcher interface {
 	// getPluginSupportedMetric returns the supported metrics for each plugin
 	// https://github.com/aws/amazon-cloudwatch-agent/blob/6451e8b913bcf9892f2cead08e335c913c690e6d/translator/translate/metrics/config/registered_metrics.go
 	getPluginSupportedMetric() map[string]struct{}
+	setEnv(env *environment.MetaData)
 }
 
-type baseMetricValueFetcher struct{}
+type baseMetricValueFetcher struct {
+	Env *environment.MetaData
+}
 
-func (f *baseMetricValueFetcher) fetch(namespace, metricName string, metricSpecificDimensions []types.Dimension, stat Statistics) (MetricValues, error) {
+func (f *baseMetricValueFetcher) getEnv() *environment.MetaData {
+	return f.Env
+}
+
+func (f *baseMetricValueFetcher) setEnv(env *environment.MetaData) {
+	f.Env = env
+}
+
+func (f *baseMetricValueFetcher) getInstanceIdDimension() types.Dimension {
 	ec2InstanceId := test.GetInstanceId()
-	instanceIdDimension := types.Dimension{
+
+	//TODO For now they can stay. Later host metrics fetchers might need to be flexible on how to get instance Id
```

Comment:
IMO, it should be the same considers its both EC2.

Reply:
I don't think so? For plain ec2 tests we are creating an ec2 instance, ssh into it, and run both test and agent on that created ec2 instance. For ecs tests ec2 instances exist but we aren't ssh'ing into it.